common/views/components/ImageViewer/ImageViewer.js

```diff
@@ -68,207 +84,112 @@ const ImageWrapper = styled.div`
   }
 `;
 
-function getTileSources(data) {
-  return [
-    {
-      '@context': 'http://iiif.io/api/image/2/context.json',
-      '@id': data['@id'],
-      height: data.height,
-      width: data.width,
-      profile: ['http://iiif.io/api/image/2/level2.json'],
-      protocol: 'http://iiif.io/api/image',
-      tiles: [
-        {
-          scaleFactors: [1, 2, 4, 8, 16, 32],
-          width: 400,
-        },
-      ],
-    },
-  ];
-}
 type ImageViewerProps = {|
   id: string,
-  src: string,
-  srcSet: string,
   width: number,
   height?: number,
   infoUrl: string,
   lang: ?string,
+  alt: string,
   tabbableControls: boolean,
+  urlTemplate: IiifUriOpts => string,
+  presentationOnly?: boolean,
 |};
 
 const ImageViewer = ({
   id,
   width,
   height,
   lang,
+  alt,
   infoUrl,
-  src,
-  srcSet,
   tabbableControls,
+  urlTemplate,
+  presentationOnly,
 }: ImageViewerProps) => {
+  const [showViewer, setShowViewer] = useState(false);
   const [imageLoading, setImageLoading] = useState(false);
-  const [isLoadingViewer, setIsLoadingViewer] = useState(false);
-  const [viewer, setViewer] = useState(null);
-  const [isError, setIsError] = useState(false);
-  const zoomStep = 0.5;
+  const [imageSrc, setImageSrc] = useState(
+    urlTemplate && urlTemplate({ size: '640,' })
+  );
+
+  const [imageSrcSet, setImageSrcSet] = useState(
+    urlTemplate &&
+      imageSizes(2048)
+        .map(width => {
+          const urlString = urlTemplate({
+            size: `${width},`,
+          });
+
+          return urlString && `${urlString} ${width}w`;
+        })
+        .join(',')
+  );
+  const [rotation, setRotation] = useState(0);
+
+  useEffect(() => {
+    urlTemplate &&
+      setImageSrc(urlTemplate({ size: '640,', rotation: rotation }));
+    urlTemplate &&
+      setImageSrcSet(
+        imageSizes(2048)
+          .map(width => {
+            const urlString = urlTemplate({
+              size: `${width},`,
+              rotation: rotation,
+            });
+            return urlString && `${urlString} ${width}w`;
+          })
+          .join(',')
+      );
+  }, [infoUrl, rotation]);
 
   function routeChangeStart(url: string) {
     if (window.history.state.as !== url) {
       setImageLoading(true);
     }
   }
+
+  const escapeCloseViewer = ({ keyCode }: KeyboardEvent) => {
```

Comment:
Would this make sense be a `onKeyDown`  or `keyboardShortcutsListener` to make sure we don't proliferate the listeners?